pkg/prometheus/rules_test.go

```diff
@@ -19,10 +19,114 @@ import (
 	"testing"
 
 	monitoringv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
-
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
+func TestEnforcedNamespaceLabelRule(t *testing.T) {
+	type testCase struct {
+		PromRuleSpec                   monitoringv1.PrometheusRuleSpec
+		PromSpecEnforcedNamespaceLabel string
+		PromRuleNamespace              string
+
+		Expected string
+	}
+
+	testcases := []testCase{
+		{
+			PromRuleSpec: monitoringv1.PrometheusRuleSpec{
+				Groups: []monitoringv1.RuleGroup{
+					{
+						Name: "alertname",
```

Comment:
so far we only have alert rule examples here, can we also make sure to test this with recording rules?